unit_tests/src/test_environment_variables.cpp

```diff
@@ -21,39 +22,54 @@ TEST_CASE("environment variables")
         // Assert the working case:
         // Given a large enough buffer, the environment variable will be coppied into the buffer
         constexpr std::size_t LARGE_BUFFER_SIZE = 100;
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,-warnings-as-errors)
         char large_buffer[LARGE_BUFFER_SIZE]{};
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
         REQUIRE(0 == capture_string_property(env_var_key.c_str(), large_buffer, LARGE_BUFFER_SIZE));
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
         REQUIRE(strnlen(large_buffer, LARGE_BUFFER_SIZE) == env_var_val.size());
```

Comment:
Each one of these blocks could live in their own section.

Not required, but could improve the test by allowing users to pick which ones are run.